enumerable.rb

```diff
@@ -0,0 +1,88 @@
+module Enumerable
+  def my_each
+    count = 0
+    while count < self.size
+      yield(self[count])
+      count += 1
+    end
+
+  def my_each_width_index
+    index = 0
+    for element in self
+      yield(self[element], element)
+      index += 1
+  end
+
+  def my_select
+    result= []
+    self.my_each{|element|
+      if yield(element)
+        result << element
+      end
+    }  
+  end
+
+  def my_all?
+    result = false
+    self.my_each{|element| yield(element) ? result = true : result= false}
+    result
+  end
+
+  def my_any?
+    result = false
+    self.my_each{|element| result = true if yield(element)}
+  end
+
+  def my_none?
+    result = true
+    self.my_each{|element| result = false if  yield (element)}
+  end
+
+  def count

```

Comment:
`count` method misses a functionality. It should return the size if we `arr.count` which in this case we haven't passsed a `block`. So we need to check if we have a block first, you can use:
```ruby
def count
  return self.size unless block.given?
  ## rest of code
end
```

Also you are overriding the default `count` method in here. Please rename it to `my_count`

Reply:
done